driver/pelican/types/thermSchedule.go

```diff
@@ -0,0 +1,252 @@
+	if errs != nil {
+		return nil, fmt.Errorf("Failed to retrieve schedule settings for thermostat %v: %v", thermostatID, errs)
+	}
+	var result settingsRequest
+	decoder := json.NewDecoder(resp.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return nil, fmt.Errorf("Failed to decode schedule settings for thermostat %v: %v", thermostatID, decodeError)
+	}
+	return &result.Userdata, nil
+}
+
+func getScheduleByDay(dayOfWeek int, epnum float64, sitename, thermostatID string, cookie *http.Cookie, timezone *time.Location) (*ThermostatDaySchedule, error) {
+	// Construct Request URL for Thermostat Schedule by Day of Week
+	var requestURL bytes.Buffer
+	requestURL.WriteString(fmt.Sprintf("https://%s.officeclimatecontrol.net/thermDayEdit.cgi?section=json&nodename=", sitename))
+	requestURL.WriteString(thermostatID)
+	requestURL.WriteString("&epnum=")
+	requestURL.WriteString(fmt.Sprintf("%.0f", epnum))
+	requestURL.WriteString("&dayofweek=")
+	requestURL.WriteString(strconv.Itoa(dayOfWeek))
+
+	// Make Request, Decode into Response Struct
+	resp, _, errs := gorequest.New().Get(requestURL.String()).Type("form").AddCookie(cookie).End()
+	if errs != nil {
+		return nil, fmt.Errorf("Failed to retrieve schedule for thermostat %v on day of week %v: %v", thermostatID, dayOfWeek, errs)
+	}
+	var result scheduleRequest
+	decoder := json.NewDecoder(resp.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return nil, fmt.Errorf("Failed to decode schedule for thermostat %v on day of week %v: %v", thermostatID, dayOfWeek, decodeError)
+	}
+
+	// Transfer Response Struct Data into return struct
+	var daySchedule ThermostatDaySchedule
+	for _, block := range result.ClientData.SetTimes {
+		var returnBlock ThermostatBlockSchedule
+		returnBlock.CoolSetting = block.CoolSetting
+		returnBlock.HeatSetting = block.HeatSetting
+		returnBlock.System = block.System
+
+		if rruleTime, rruleError := convertTimeToRRule(dayOfWeek, block.StartValue, timezone); rruleError != nil {
+			return nil, fmt.Errorf("Failed to convert time in string format %v to rrule format: %v", block.StartValue, rruleError)
+		} else {
+			returnBlock.Time = rruleTime
+		}
+
+		daySchedule.Blocks = append(daySchedule.Blocks, returnBlock)
+	}
+	return &daySchedule, nil
+}
+
+func convertTimeToRRule(dayOfWeek int, blockTime string, timezone *time.Location) (string, error) {
+	timeSlice := strings.Split(blockTime, ":")
+	hour, hourErr := strconv.Atoi(timeSlice[0])
+	if hourErr != nil {
+		return "", fmt.Errorf("Failed to convert hour value of type string to type int: %v", hourErr)
+	}
+	if timeSlice[2] == "PM" {
+		hour += 12
+		if hour == 24 {
+			hour = 0
+		}
+	}
+	minute, minuteErr := strconv.Atoi(timeSlice[1])
+	if minuteErr != nil {
+		return "", fmt.Errorf("Failed to convert minute value of type string to type int: %v", minuteErr)
+	}
+
+	rruleSched, _ := rrule.NewRRule(rrule.ROption{
```

Comment:
Is there a reason you don't check the error value returned by `NewRRule`?

Reply:
I think that was a remnant of me testing the rruleSched initially, I added a variable to check the error!